app/controllers/alert_requests_controller.rb

```diff
@@ -0,0 +1,114 @@
+class AlertRequestsController < ApplicationController
+  layout 'alert_request'
+  before_action :find_partner
+
+  def new
+    @alert_request = AlertRequest.new(
+      partner_id: @partner_id, 
+      tracking_source: @source,
+      tracking_id: @tracking,
+      phone_type: 'mobile',
+    )
+
+    # do we need this?
+    # if @alert_request.partner.primary?
+    #   @alert_request.opt_in_email = true
+    # else
+    #   if @alert_request.partner.rtv_email_opt_in?
+    #     @alert_request.opt_in_email = true
+    #   end
+    #   if @alert_request.partner.partner_email_opt_in?
+    #     @alert_request.partner_opt_in_email = true
+    #   end
+    # end
+  
+    set_up_locale
+    @registrant = OpenStruct.new
+    @question_1 = @alert_request.question_1
+    @question_2 = @alert_request.question_2
+  end
+
+  def create
+    @alert_request = AlertRequest.create(
+      alert_request_params.merge(
+        uid: SecureRandom.hex(20),
+        state: @home_state,
+        partner_id: @partner_id,
+      )
+    )
+    @question_1 = @alert_request.question_1
+    @question_2 = @alert_request.question_2
+    update_partner_questions
+
+    if @alert_request.save
+      redirect_to alert_request_path(@alert_request)
+    else
+      @registrant = OpenStruct.new
+      render :new
+    end
+  end
+
+  def show
+    @alert_request = AlertRequest.find_by_param(params[:id])
+  end
+
+  private
+
+  def alert_request_params
+    @alert_request_params ||= params.require(:alert_request).permit(
+      'first',
+      'last',
+      'suffix',
+      'address',
+      'city',
+      'zip',
+      'date_of_birth_month',
+      'date_of_birth_day',
+      'date_of_birth_year',
+      'tracking_source',
+      'tracking_id',
+      'phone',
+      'phone_type',
+      'email',
+      'survey_answer_1',
+      'survey_answer_2',
+      # ?
+      # 'opt_in_email',
+      # 'opt_in_sms',
+      # 'partner_opt_in_email',
+      # 'partner_opt_in_sms',
+      # TODO: extra tracking params
+    )
+  end
+
+  def find_partner
+    @partner = Partner.find_by_id(params[:partner]) || Partner.find(Partner::DEFAULT_ID)
+    @partner_id = @partner.id
+    set_params
+  end
+
+  def set_params
```

Comment:
I think we should pretty much allow any field on the form (except opt-ins and survey answers) to be set via query string on the landing page.

Reply:
implemented via `new_alert_request_params`